src/installer.ts

```diff
@@ -65,22 +65,28 @@ async function execOpenSSLVersion() {
 }
 
 async function getOpenSSLVersion(version = ''): Promise<string> {
-  if (version === '') {
-    version = await execOpenSSLVersion();
-  }
+  const specificVersionViaInput = /^\d{1,3}\.\d{1,3}$/.test(version);
 
-  // typical version string looks like 'OpenSSL 1.0.1e-fips 11 Feb 2013'
-  const openSSLVersionPattern = /^([^\s]*)\s([0-9]+\.[0-9]+)/;
-  const match = openSSLVersionPattern.exec(version);
+  if (!specificVersionViaInput) {
```

Comment:
Can we invert this conditional and just short circuit?

```suggestion
  if (specificVersionViaInput) {
    return version;
  }
```

Then everything below doesn't really have to change (and it's easier for me to reason about :P )

Reply:
Changed to early return.